kin/blockchain/builder.py

```diff
@@ -3,33 +3,31 @@
 from kin_base.builder import Builder as BaseBuilder
 from kin_base.keypair import Keypair
 from kin_base.memo import NoneMemo
-from kin_base.exceptions import StellarAddressInvalidError
-
-from .utils import is_valid_address, is_valid_secret_key
 
 
 class Builder(BaseBuilder):
     """
     This class overrides :class:`kin_base.builder` to provide additional functionality.
+    # TODO: maybe merge this with kin-base builder
     """
 
-    def __init__(self, network, horizon, fee, secret=None, address=None):
-        if secret:
-            if not is_valid_secret_key(secret):
-                raise ValueError('invalid secret key')
-            address = Keypair.from_seed(secret).address().decode()
-        elif address:
-            if not is_valid_address(address):
-                raise StellarAddressInvalidError('invalid address: {}'.format(address))
-        else:
-            raise Exception('either secret or address must be provided')
+    # TODO: make seed optional (need to change kin_base)
+    def __init__(self, network_name, horizon, fee, secret):
+        """
+        Create a new transaction builder
+        :param str network_name: The name of the network
+        :param Kin.Horizon horizon: The horizon instance to use
+        :param int fee: Fee for the transaction
+        :param str secret: The seed to be used
+        """
 
-        # call baseclass constructor to init base class variables
-        super(Builder, self).__init__(secret=secret, address=address, sequence=1, fee=fee)
+        # call base class constructor to init base class variables
+        # sequence is one since its get overridden later
+        super(Builder, self).__init__(secret=secret, sequence=1, fee=fee)
 
         # custom overrides
 
-        self.network = network
+        self.network = network_name
```

Comment:
why not keep the same name `self.network_name = network_name`

Reply:
"network" is used in kin-base, so I cant only change it here